File: LeetCode_Solutions/2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        int i=0;
        while(k>=0){
            if(k>=chalk[i]){
                k-=chalk[i];
                i++;
            }
            else{
                return i;
            }
            if(i==chalk.length){
                i=0;
            }
            
           // System.out.print(k+" ");
           // System.out.print(i+" ");
        }
        return k;
    }
}

File: LeetCode_Solutions/2006-find-the-student-that-will-replace-the-chalk/find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        n=len(chalk)
        sum1=0
        for i in range(n):
            sum1+=chalk[i]
            if sum1>k:
                return i
        return self.chalkReplacer(chalk,k%sum1)